collectors/proc.plugin/sys_class_power_supply.c

```diff
@@ -258,36 +275,38 @@ int do_sys_class_power_supply(int update_every, usec_t dt) {
             for(pr = ps->property_root; pr && !read_error; pr = pr->next) {
                 struct ps_property_dim *pd;
                 for(pd = pr->property_dim_root; pd; pd = pd->next) {
-                    char buffer[30 + 1];
+                    if(likely(!pd->always_zero)) {
```

Comment:
This, on the other hand, makes sense to be marked as likely, because most of the systems I have seen have significantly more correct (not-always-zero) dimensions.